puni.el

```diff
@@ -2376,6 +2418,123 @@ with it."
 
       (goto-char pt-to-restore))))
 
+;;; Wrapping
+
+(defun puni--wrap-region (beg end beg-delim end-delim)
+  "Wrap region between BEG and END with BEG-DELIM and END-DELIM.
+The indentation of the region is adjusted to make it the same
+like before wrapping.  BEG and END are integers, not markers."
+  (let ((beg (min beg end))
+        (end (max beg end)))
+    (save-excursion
+      (goto-char end)
+      (insert end-delim)
+      (goto-char beg)
+      (insert beg-delim)
+      (puni--reindent-region
+       (+ beg (length beg-delim))
+       (+ end (length beg-delim) (length end-delim))
+       (puni--column-of-position beg))))
+  (if (< beg end)
+      (goto-char (+ beg (length beg-delim)))
+    (goto-char end)
+    (puni-strict-forward-sexp)
+    (goto-char (- (point) (length end-delim)))))
+
+;;;###autoload
+(defun puni-wrap-next-sexps (n beg-delim end-delim)
+  "Wrap next N S-expressions with BEG-DELIM and END-DELIM.
+- If N is a positive integer, wrap N sexps after the point.
+- If N is `to-end', wrap sexps from the point to the end of
+  current list.
+- If N is a negative integer, wrap N sexps before the point.
+- If N is `to-beg', wrap sexps from the point to the beginning of
+  current list.
+- If N is `region', and there's an active region, wrap the region
+  instead, otherwise throw an error."
+  (puni--set-undo-position)
+  (if (eq n 'region)
+      (puni--wrap-region (point) (mark) beg-delim end-delim)
+    (let* ((n (cond ((eq n 'to-end) most-positive-fixnum)
+                    ((eq n 'to-beg) most-negative-fixnum)
+                    ((numberp n) n)
+                    (t (user-error
+                        "expected 'to-end, 'to-beg, 'region, or integer as N, got: %S"
+                        n))))
+           (beg (save-excursion
+                  (if (>= n 0)
+                      (puni--forward-blanks)
+                    (puni--backward-blanks))
+                  (point)))
+           ((end (save-excursion
```

Comment:
Extra `()` found here ;) Maybe you are experiment with `puni-wrap-curly` when writing this?

Reply:
haha, yeah, sorry :sweat_smile: